### PR TITLE
Mock Auth Page & Button

### LINKED ISSUE #53351: [FE] (Mock Authentication) - Add mock authentication button
## Description
Create a mock authentication button that can be enabled on the Unified Sign in Page as well as on the Sign in Modal (in dev & below environments).

## Tasks
- [x] Create mock authentication button that generates the proper URL for Sign in Service
- [ ] ~~Ensure button does not display for Production (only staging & below environments)~~
- [x] Button should display on /sign-in/mocked-auth 
- [x] Clicking the button takes you to the `<api-env>/v0/sign_in/authorize?client_id=vamock` API URL
- [x] Ensure the button color is different than any other CSP

### PR DESCRIPTION
**Pull request by ianmagenta**
## Summary
- Added mock auth page at /sign-in/mocked-auth (this exists in all environments)
- Added mock auth button that redirects to `<api-env>/v0/sign_in/authorize?client_id=vamock`
- Made the button a different color than any other CSP
- Created tests for the above

## Related issue(s)
Closes https://github.com/department-of-veterans-affairs/va.gov-team/issues/53351

## Screenshots
image

## What areas of the site does it impact?
This adds components to `src/applications/login/` and adds a child-route to `/sign-in`.

## Acceptance criteria
- [x] Create mock authentication button that generates the proper URL for Sign in Service
- [x] Button should display on /sign-in/mocked-auth 
- [x] Clicking the button takes you to the `<api-env>/v0/sign_in/authorize?client_id=vamock` API URL
- [x] Ensure the button color is different than any other CSP

### CODE REVIEW

**Review by rmessina1010 — APPROVED** [review]
LGTM

File: src/applications/login/containers/MockAuth.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    <section className="login">
+      <div className="container">
+        <div className="container">
+          <div className="row">
+            <div className="columns small-12">
+              <h1
+                id="signin-signup-modal-title"
+                className="vads-u-margin-top--2 vads-u-color--gray-dark medium-screen:vads-u-margin-top--1 medium-screen:vads-u-margin-bottom--2"
+              >
+                Mocked Auth
```
Comment (asg5704) [inline src/applications/login/containers/MockAuth.jsx:16]: Change to `Mock Authentication`

File: src/applications/login/containers/MockAuth.jsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+              </h1>
+            </div>
+          </div>
+          <div className="row">
+            <div className="columns small-12" id="sign-in-wrapper">
+              <MockAuthButton />
+            </div>
+          </div>
+          <LoginInfo />
```
Comment (asg5704) [inline src/applications/login/containers/MockAuth.jsx:25]: You can remove the <LoginInfo /> component.  Replace with:

Paragraph: 
Mocked authentication is only available in the development and localhost environments. Read more about [mock authentication](#) and how to use it on GitHub.

Once we have mock authentication documentation we can come back and update the URL.

File: src/applications/login/components/MockAuthButton.jsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    <button
+      type="button"
+      aria-label="Mock Authentication"
+      className="usa-button mauth-button vads-u-margin-y--1p5 vads-u-padding-y--2"
+      onClick={() => {
+        const environ =
+          typeof process !== 'undefined'
+            ? process.env.BUILDTYPE
+            : environment.BUILDTYPE;
+        window.location.replace(
```
Comment (asg5704) [inline src/applications/login/components/MockAuthButton.jsx:16]: We need to keep the window History for the session (ie when a person hits the back button they land where they're supposed to).  You can change this to `window.location = <url>`

File: src/applications/login/components/MockAuthButton.jsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+          `${
+            ENVIRONMENT_CONFIGURATIONS[environ].API_URL
```
Comment (asg5704) [inline src/applications/login/components/MockAuthButton.jsx:18]: Take a look at the environment utilities in `platform/utilities/environment/index.js`

File: src/applications/login/routes.jsx [hunk 5]
```diff
@@ -1,10 +1,17 @@
...
 const routes = {
   path: '/',
   component: SignInWrapper,
   indexRoute: { component: SignInApp },
+  childRoutes: [
+    {
+      path: 'mocked-auth',
+      component: MockAuth,
+    },
+  ],
```
Comment (asg5704) [inline src/applications/login/routes.jsx:14]: Let's have `childRoutes` conditionally be created for dev & localhost environments.

File: src/platform/site-wide/user-nav/sass/user-nav.scss [hunk 6]
```diff
@@ -150,6 +150,14 @@ span.sidelines {
       }
     }
 
+    &.mauth-button {
+      background: $color-gray;
+
+      &:hover {
+        background: $color-gray-dark;
+      }
+    }
```
Comment (asg5704) [inline src/platform/site-wide/user-nav/sass/user-nav.scss:159]: 🚀

**Review by asg5704 — CHANGES_REQUESTED** [review]
Only a few changes needed. Looks really good otherwise, great job! 😎

File: src/applications/login/components/MockAuthButton.jsx (hunk 4 above)
Comment (ianmagenta) [reply src/applications/login/components/MockAuthButton.jsx:18]: Funny enough, the original version of this used the `API_URL` from that file. Unfortunately, I couldn't use that in conjunction with the tests I wrote (which change `process.env.BUILDTYPE` to see if the URL gets set properly in different environments). This was the best way I could think of to have the button work both during the tests and when deployed. If there's a better way, please feel free to let me know!

**Comment by ianmagenta** [comment]
@asg5704 - Thanks for the feedback, I just need to update the tests and this should be good to go! I'll ping you once that's complete!

**Comment by ianmagenta** [comment]
@asg5704 - Tests have been pushed up, this should be good for re-review!

File: src/applications/login/routes.jsx [hunk 7]
```diff
@@ -1,10 +1,21 @@
...
   indexRoute: { component: SignInApp },
+  childRoutes:
+    environment.isDev() || environment.isLocalhost()
+      ? [
+          {
+            path: 'mocked-auth',
+            component: MockAuth,
+          },
+        ]
+      : [],
```
Comment (asg5704) [inline src/applications/login/routes.jsx:18]: 🔥

**Review by asg5704 — APPROVED** [review]
LGTM.  Once Afia's work make it in we should be able to modify the `onClick` to a utility function. Great job.

**Comment by asg5704** [comment]
@ianmagenta You can merge this in 👀

**Comment by ianmagenta** [comment]
@asg5704 - Merged!

**Pull request merged**